core/src/main/java/org/elasticsearch/index/query/BoostingQueryBuilder.java

```diff
@@ -81,25 +126,87 @@ protected void doXContent(XContentBuilder builder, Params params) throws IOExcep
         if (negativeQuery == null) {
```

Comment:
I had them in validate() before, but the problem like in #11629 is that we need to support inner clauses that parse to `null` like `"positive": { "constant_score": { "filter": {}}}`, so if we have these checks in validate() we will throw errors on queries the DSL currently accepts.
I wonder why this behaviour was introduced in the first place, maybe there is a deeper reason for that, otherwise we should try and get rid of this but I think that's out of scope for this refactoring for now. Do you agree?